src/app/util/CHIPDeviceCallbacksMgr.h

```diff
@@ -81,7 +83,7 @@ class DLL_EXPORT CHIPDeviceCallbacksMgr
         Callback::Cancelable * ca = &queue;
         while (ca != nullptr && ca->mNext != &queue)
         {
-            if (*reinterpret_cast<T *>(&ca->mNext->mInfoPtr) == info)
+            if (*reinterpret_cast<T *>(ca->mNext->mInfoPtr) == info)
```

Comment:
This is unsafe as sizeof T is not checked to fit within mInfoPtr. Also, T is not actually mutated, so it doesn't need to be a ref.

Need to have instead:
```
template <typename T>
    CHIP_ERROR GetCallback(T info, Callback::CallbackDeque & queue, Callback::Cancelable ** callback)
```
It is still problematic since we are *expecting* here that the mInfoPtr has sizeof(T) bytes in it, but it may not. If you mix different types of info, some will "read past the end of the buffer".

mInfoPtr needs to have a struct that tells you the size, so that you skip over any entry of the incorrect size, otherwise there is a buffer overrun on read.